opensearch/examples/index_template.rs

```diff
@@ -0,0 +1,139 @@
+    client.indices().put_index_template(IndicesPutIndexTemplateParts::Name("books"))
+    .body(json!({
+        "index_patterns": ["books-*"],
+        "template": {
+            "settings": {
+                "index": {
+                    "number_of_shards": 3,
+                    "number_of_replicas": 0
+                }
+            },
+            "mappings": {
+                "properties": {
+                    "title": { "type": "text" },
+                    "author": { "type": "text" },
+                    "published_on": { "type": "date" },
+                    "pages": { "type": "integer" }
+                }
+            }
+        }
+    }))
+    .send().await?;
+    // Now, when you create an index that matches the `books-*` pattern, OpenSearch will automatically apply the template's settings and mappings to the index.
+    // Let's create an index named `books-nonfiction` and verify that its settings and mappings match those of the template:
+    client.indices().create(IndicesCreateParts::Index("books-nonfiction")).send().await?;
+    client.indices().get(IndicesGetParts::Index(&["books-nonfiction"])).send().await?;
+    // If multiple index templates match the index's name, OpenSearch will apply the template with the highest priority. The following example creates two index templates named `books-*` and `books-fiction-*` with different settings:
+    client.indices().put_index_template(IndicesPutIndexTemplateParts::Name("books"))
+    .body(json!({
+        "index_patterns": ["books-*"],
+        "priority": 0,
+        "template": {
+            "settings": {
+                "index": {
+                    "number_of_shards": 3,
+                    "number_of_replicas": 0
+                }
+            }
+        }
+    }))
+    .send().await?;
+
+    client.indices().put_index_template(IndicesPutIndexTemplateParts::Name("books-fiction"))
+    .body(json!({
+        "index_patterns": ["books-fiction-*"],
+        "priority": 1,
+        "template": {
+            "settings": {
+                "index": {
+                    "number_of_shards": 1,
+                    "number_of_replicas": 1
+                }
+            }
+        }
+    }))
+    .send().await?;
+    // When we create an index named `books-fiction-romance`, OpenSearch will apply the `books-fiction-*` template's settings to the index:
+    client.indices().create(IndicesCreateParts::Index("books-fiction-romance")).send().await?;
+    client.indices().get(IndicesGetParts::Index(&["books-fiction-romance"])).send().await?;
+    //Composable index templates are a new type of index template that allow you to define multiple component templates and compose them into a final template. The following example creates a component template named `books_mappings` with default mappings for indices of the `books-*` and `books-fiction-*` patterns:
```

Comment:
Add space after //

Reply:
@dblock Check if I've got it right? If inserted exactly after "//" it does not look very nice

Reply:
diff still stays `//Composable`

I believe we have an auto-formatter for rust?

Reply:
Yes, sorry, corrected it!
I used cargo fmt as I am having problems with beautify